.chloggen/jaegerremotesampling_replace_thrift-gen_with_proto-gen.yaml

```diff
@@ -0,0 +1,19 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: jaegerreceiver,jaegerremotesamplingextension
```

Comment:
After adding a featuregate to the `jaegerremotesamplingextension` the module worked fine, but it was not possible to compile the contrib client. Since otelcontribcol used 1.48.

<details>
<summary>Log</summary>

```
cd ./cmd/otelcontribcol && GO111MODULE=on CGO_ENABLED=0 go build -trimpath -o ../../bin/otelcontribcol_linux_amd64 \
	-tags "" .
# github.com/open-telemetry/opentelemetry-collector-contrib/receiver/jaegerreceiver
../../receiver/jaegerreceiver/trace_receiver.go:174:43: cannot use (*notImplementedConfigManager)(nil) (value of type *notImplementedConfigManager) as configmanager.ClientConfigManager value in variable declaration: *notImplementedConfigManager does not implement configmanager.ClientConfigManager (wrong type for method GetSamplingStrategy)
		have GetSamplingStrategy(context.Context, string) (*"github.com/jaegertracing/jaeger/thrift-gen/sampling".SamplingStrategyResponse, error)
		want GetSamplingStrategy(context.Context, string) (*api_v2.SamplingStrategyResponse, error)
../../receiver/jaegerreceiver/trace_receiver.go:281:74: cannot use &notImplementedConfigManager{} (value of type *notImplementedConfigManager) as configmanager.ClientConfigManager value in argument to httpserver.NewHTTPServer: *notImplementedConfigManager does not implement configmanager.ClientConfigManager (wrong type for method GetSamplingStrategy)
		have GetSamplingStrategy(context.Context, string) (*"github.com/jaegertracing/jaeger/thrift-gen/sampling".SamplingStrategyResponse, error)
		want GetSamplingStrategy(context.Context, string) (*api_v2.SamplingStrategyResponse, error)
# github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/zipkin/zipkinv1
../../pkg/translator/zipkin/zipkinv1/thrift.go:27:47: not enough arguments in call to jaegerzipkin.DeserializeThrift
	have ([]byte)
	want (context.Context, []byte)
make: *** [Makefile:287: otelcontribcol] Fehler 1
```
</details>

Therefore, I have increased the version there the same way and provided another [feature gate](https://github.com/open-telemetry/opentelemetry-collector/blob/26c150c8bee385184d26669701560f99245bb16d/featuregate/README.md). They work independently of each other.

Reply:
cc @jpkrohling @zcross @MovieStoreGuy